### PR TITLE
Add InitWithClient and deprecate InitClient because we don't want to copy Client's field sync.RWMutex by value

### PR DESCRIPTION
**Pull request by mandyh2018**
Init proxy client using pointer because we don't want to copy its filed sync.RWMutex by value.

### CODE REVIEW

**Comment by hfwang** [comment]
I think this may not be allowed actually since it changes the API surface...

**Comment by mandyh2018** [comment]
hmm, I can also add a new API.

**Comment by broady** [comment]
Can you tell us about the motivation behind this change? Is there a bug?

**Comment by Carrotman42** [comment]
Copying a sync.RWMutex is not generally safe, although I think copying an unused (i.e. zero-value) RWMutex can't be noticed by the internal code used to detect this copying. In any case this was definitely an oversight when I wrote this API and the API should be either changed or deprecated/deleted-eventually.

**Comment by kurtisvg** [comment]
broady@ and I discussed this today and the conclusion was that we should defiantly make this change, but play it safe and create a new function and deprecate the old one.

**Comment by mandyh2018** [comment]
@kurtisvg Thanks! Is it part of your plan? Or I will do it later. :-)

**Comment by kurtisvg** [comment]
@mandyh2018 If you would like to make the change you are welcome to, otherwise I will update this PR sometime before the next release.

**Comment by mandyh2018** [comment]
broady@ are we okay with changing the existing API? I thought we want to add a new one instead.

**Comment by mandyh2018** [comment]
kurtisvg@ Are you working on this? If not, I will add a new API this week. :-)

**Comment by kurtisvg** [comment]
My plan is to just add a new function (likely `InitWithClient`) and mark the old one with a deprecated warning. I'm hoping to do it sometime this week, but please feel free to do it sooner if you have time :)

**Comment by mandyh2018** [comment]
Let me work on this then. I have some time today and tomorrow. :-) So I just close this pull request and send a new one?  kurtisvg@

**Comment by kurtisvg** [comment]
You can just update this PR with the changes

**Comment by mandyh2018** [comment]
I updated this PR, please let me know if any comment. Thanks!

File: proxy/proxy/dial.go [hunk 1]
```diff
@@ -80,12 +81,22 @@ func Init(auth *http.Client, connset *ConnSet, dialer Dialer) {
 
 // InitClient is similar to Init, but allows you to specify the Client
 // directly.
+
+// Deprecated: Client contains sync.RWMutex, which should not be copied by value.
```
Comment (broady) [inline proxy/proxy/dial.go]: just "Deprecated: Use InitWithClient instead." i think.

Comment (mandyh2018) [reply proxy/proxy/dial.go]: Done. Thx!

**Comment by mandyh2018** [comment]
Thank you! I will wait until I become the authorized user?

**Comment by mandyh2018** [comment]
I can see the warning:
Merging is blocked
The base branch restricts merging to authorized users. 
You’re not authorized to merge this pull request.

**Comment by broady** [comment]
Thanks! Merged.

**Comment by mandyh2018** [comment]
Thank you!

**Pull request merged**